contracts/StakeManager.sol

```diff
@@ -377,21 +446,22 @@ contract StakeManager is Ownable {
             _mintMP(account, iEpoch.startTime + EPOCH_SIZE, iEpoch);
             uint256 userSupply = account.balance + account.totalMP;
             uint256 userEpochReward = Math.mulDiv(userSupply, iEpoch.epochReward, iEpoch.totalSupply);
-
             userReward += userEpochReward;
             iEpoch.epochReward -= userEpochReward;
             iEpoch.totalSupply -= userSupply;
+            if (iEpoch.totalSupply == 0) {
+                pendingReward -= iEpoch.epochReward;
+                delete epochs[userEpoch];
+            }
```

Comment:
Hm.. This seems a bit odd.
We're subtracting `iEpoch.epochReward` from `pendingReward`, and we're also subtracting `userReward` from `pendingReward` in line `459`.

Shouldn't we only subtract the `userReward` from `pendingReward`? And shouldn't we only do this once, not twice? 

Also, this changes causes the rule to fail that `subOfEpochRewards == pendingReward`
Need to check this.